tests/common/snappi/common_helpers.py

```diff
@@ -436,68 +448,113 @@ def config_wred(host_ans, kmin, kmax, pmax, profile=None):
+            gmin_cmd = 'sudo ip netns exec %s ecnconfig -p {} -gmin {}' % asic_value
+            if asic_type == 'broadcom':
+                try:
+                    host_ans.shell('bcmcmd -n {} "BCMSAI credit-watchdog disable"'.format(asic_value))
+                except Exception:
+                    host_ans.shell('bcmcmd -n {} "BCMSAI credit-watchdog disable"'.format(asic_value[-1]))
+
+        if kmin > kmin_old:
+            host_ans.shell(gmax_cmd.format(p, kmax))
+            host_ans.shell(gmin_cmd.format(p, kmin))
         else:
-            host_ans.shell('sudo ecnconfig -p {} -gmin {}'.format(p, kmin))
-            host_ans.shell('sudo ecnconfig -p {} -gmax {}'.format(p, kmax))
+            host_ans.shell(gmin_cmd.format(p, kmin))
+            host_ans.shell(gmax_cmd.format(p, kmax))
 
     return True
 
 
-def enable_ecn(host_ans, prio):
+def enable_ecn(host_ans, prio, asic_value='None'):
     """
     Enable ECN marking on a priority
+
     Args:
         host_ans: Ansible host instance of the device
         prio (int): priority
+        asic_value: asic value of the host
+
     Returns:
         N/A
     """
-    host_ans.shell('sudo ecnconfig -q {} on'.format(prio))
+    if asic_value == 'None':
+        host_ans.shell('sudo ecnconfig -q {} on'.format(prio))
+    else:
+        host_ans.shell('sudo ip netns exec {} ecnconfig -q {} on'.format(asic_value, prio))
 
 
-def disable_ecn(host_ans, prio):
+def disable_ecn(host_ans, prio, asic_value='None'):
     """
     Disable ECN marking on a priority
+
     Args:
         host_ans: Ansible host instance of the device
         prio (int): priority
+        asic_value: asic value of the host
+
     Returns:
         N/A
     """
-    host_ans.shell('sudo ecnconfig -q {} off'.format(prio))
+    if asic_value == 'None':
+        host_ans.shell('sudo ecnconfig -q {} off'.format(prio))
+    else:
+        asic_type = str(host_ans.facts["asic_type"])
+        host_ans.shell('sudo ip netns exec {} ecnconfig -q {} off'.format(asic_value, prio))
+        if asic_type == 'broadcom':
+            try:
+                host_ans.shell('bcmcmd -n {} "BCMSAI credit-watchdog enable"'.format(asic_value))
+            except Exception:
+                host_ans.shell('bcmcmd -n {} "BCMSAI credit-watchdog enable"'.format(asic_value[-1]))
 
 
-def config_buffer_alpha(host_ans, profile, alpha_log2):
+def config_buffer_alpha(host_ans, profile, alpha_log2, asic_value='None'):
     """
     Configure buffer threshold (a.k.a., alpha)
+
     Args:
         host_ans: Ansible host instance of the device
         profile (str): buffer profile name
         alpha_log2 (int): set threshold to 2^alpha_log2
+        asic_value: asic value of the host
+
     Returns:
         N/A
     """
-    host_ans.shell('sudo mmuconfig -p {} -a {}'.format(profile, alpha_log2))
+    if asic_value == 'None':
+        host_ans.shell('sudo mmuconfig -p {} -a {}'.format(profile, alpha_log2))
```

Comment:
It would be nice if duthost.shell can encapsulate the namespace instead of having to expand every single shell command with `ip netns exec ...`

Reply:
can you please provide detailed explanation as to how this has to be done, like an example

Reply:
@kenneth-arista  @selldinesh Maybe we can define some functions called mmuconfig(host_ans, asic_value, profile, alpha), enconfig(host_ans, asic_value, kmin, kmax, pmax)